x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/pages/coverage_overview/filters_panel.tsx

```diff
@@ -5,42 +5,188 @@
+        value,
+      });
+    },
+    [dispatch]
+  );
+
+  const setRuleSearchFilter = useCallback(
+    (value: string): void => {
+      dispatch({
+        type: 'setRuleSearchFilter',
+        value,
+      });
+    },
+    [dispatch]
+  );
+
   const handleExpandCellsFilterClick = () => setShowExpandedCells(true);
   const handleCollapseCellsFilterClick = () => setShowExpandedCells(false);
 
+  const ruleStatusFilterInitalValue = useMemo(
+    () => getInitialRuleStatusFilterOptions(filter),
+    [filter]
+  );
+  const ruleTypeFilterInitalValue = useMemo(
+    () => getInitialRuleTypeFilterOptions(filter),
+    [filter]
+  );
+
+  const [ruleStatusFilterOptions, setRuleStatusFilterOptions] = useState(
+    ruleStatusFilterInitalValue
+  );
+
+  const [ruleTypeFilterOptions, setRuleTypeFilterOptions] = useState(ruleTypeFilterInitalValue);
+
+  const handleRuleStatusFilterOnChange = useCallback(
+    (options) => {
+      setRuleStatusFilterOptions(options);
+      const formattedOptions = formatRuleFilterOptions<CoverageOverviewRuleActivity>(options);
+      setRuleStatusFilter(formattedOptions);
+    },
+    [setRuleStatusFilter]
+  );
+
+  const handleRuleTypeFilterOnChange = useCallback(
+    (options) => {
+      setRuleTypeFilterOptions(options);
+      const formattedOptions = formatRuleFilterOptions<CoverageOverviewRuleSource>(options);
+      setRuleTypeFilter(formattedOptions);
+    },
+    [setRuleTypeFilter]
+  );
+
+  const handleRuleStatusFilterOnClear = useCallback(() => {
+    handleRuleStatusFilterOnChange(ruleStatusFilterDefaultOptions);
+  }, [handleRuleStatusFilterOnChange]);
+
+  const handleRuleTypeFilterOnClear = useCallback(() => {
+    handleRuleTypeFilterOnChange(ruleTypeFilterDefaultOptions);
+  }, [handleRuleTypeFilterOnChange]);
+
+  const handleRuleSearchOnChange = useCallback(
+    ({ queryText }: { queryText: string }) => {
+      setRuleSearchFilter(queryText);
+    },
+    [setRuleSearchFilter]
+  );
+
   return (
     <EuiPanel>
       <EuiFlexGroup justifyContent="spaceBetween">
-        <EuiFlexItem grow={false}>
-          <EuiFilterGroup>
-            <EuiFilterButton
-              withNext
-              hasActiveFilters={!showExpandedCells}
-              onClick={handleCollapseCellsFilterClick}
-            >
-              {i18n.COLLAPSE_CELLS_FILTER_BUTTON}
-            </EuiFilterButton>
-            <EuiFilterButton
-              hasActiveFilters={showExpandedCells}
-              onClick={handleExpandCellsFilterClick}
-            >
-              {i18n.EXPAND_CELLS_FILTER_BUTTON}
-            </EuiFilterButton>
-          </EuiFilterGroup>
+        <EuiFlexItem>
+          <EuiFlexGroup
+            css={css`
+              flex-grow: 0;
+            `}
+          >
+            <DashboardFilterButtonComponent
+              dataTestSubj="coverageOverviewRuleStatusFilterButton"
+              title={i18n.CoverageOverviewRuleStatusFilterLabel}
+              options={ruleStatusFilterOptions}
```

Comment:
Filter names should conform with defined in the API contract, see this [comment](https://github.com/elastic/kibana/issues/158202#issuecomment-1591746049) with the suggestion and the [API contract](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/common/api/detection_engine/rule_management/coverage_overview/coverage_overview_route.ts#L69-L73).

- `RuleStatus` and all related code should be renamed to `RuleActivity`
- `RuleType` and all related code should be renamed to `RuleSource`